### PR TITLE
Deterministic order

### PR DESCRIPTION
**Pull request by RussTheAerialist**
Address the comment in #101 about sort/filter not being in a deterministic order
(not sure why the doc changes are pulling in here, but I can assure you I didn't make those changes :)

### CODE REVIEW

**Comment by t8y8** [comment]
If order is important to us -- why not just store this as a list instead of a set, seems like a lot of hoops to keep it as a set.

(OrderedDict is another option).

We'd lose `.add` as the API, but we can just alias .add to mean `.append`?

**Comment by RussTheAerialist** [comment]
Sets ensure uniqueness.  OrderedDict would be fine too.  A list would be more work since you'll have to check for uniqueness. I don't remember the exact reasons we picked set, though

**Comment by t8y8** [comment]
OrderedDict is a nice data structure to extend in the future if need be, and has a nice key->value relationship.

Actually, that's what we're exploring in #114 anyway, so maybe I can defer my psuedo-objection to that discussion.

File: docs/_includes/analytics.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<!-- Google Tag Manager -->
+<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+})(window,document,'script','dataLayer','GTM-BVCN');</script>
+<!-- End Google Tag Manager -->
```
Comment (t8y8) [inline docs/_includes/analytics.html]: Gotta rebase on top of development, this was forked from master.

@jdomingu owes me a 🍺 at this point :)

File: test/test_request_option.py [hunk 2]
```diff
@@ -89,3 +90,20 @@ def test_filter_tags_in(self):
         self.assertEqual(set(['weather']), matching_workbooks[0].tags)
         self.assertEqual(set(['safari']), matching_workbooks[1].tags)
         self.assertEqual(set(['sample']), matching_workbooks[2].tags)
+
+    def test_multiple_filter_options(self):
+        with open(FILTER_MULTIPLE, 'rb') as f:
+            response_xml = f.read().decode('utf-8')
+        # To ensure that this is deterministic, run this a few times
+        with requests_mock.mock() as m:
+            # Sometimes pep8 requires you to do things you might not otherwise do
```
Comment (t8y8) [inline test/test_request_option.py]: lol

File: docs/_includes/analytics.html (hunk 1 above)
Comment (jdomingu) [inline docs/_includes/analytics.html]: @t8y8 More like  🍻 🍻 🍻 at this point :) Sorry!

Comment (RussTheAerialist) [inline docs/_includes/analytics.html]: I thought I did rebase from development (unless I'm misunderstanding your comment)

**Comment by Talvalin** [comment]
I agree that checking the uniqueness of filters is sensible, so it's weird that the REST API accepts requests with non-unique filters (even though we then state that only the last filter of that type specified is actually used).

The other point is that I'm having a little trouble with the `test_multiple_filter_options` code. When I run tests for `test_request_option.py`, `test_multiple_filter_options` failed consistently because the ordering of the filters in the test url didn't match what the code was returning.

The test url expects the name filter to be first, then the tags filter. My understanding was that the filters are in the same order that they are specified (tags, then name), and so if I change the test url such that the tags filter is first, the test passes consistently. This makes sense.

However, the test as it is here is passing the integration tests. What gives? 😖

**Comment by t8y8** [comment]
@Talvalin yeah thatt's what this is intended to fix -- have you tried rebasing on top of @RussTheAerialist's changes and seeing if the ordering is consistent now?

File: docs/_includes/analytics.html (hunk 1 above)
Comment (t8y8) [inline docs/_includes/analytics.html]: These changes have popped up on 3 different PRs now... I thought they came form `master` but maybe they're from something else?

**Pull request merged**